lib/ansible/modules/cloud/cloudscale/cloudscale_server.py

```diff
@@ -378,12 +395,37 @@ def _update_param(self, param_key, server_info, requires_stop=False):
 
         return server_info
 
+    def _get_server_groups(self):
+        server_group_params = self._module.params['server_groups']
+        if not server_group_params:
+            return None
+
+        server_group_ids = []
+        server_groups = self._get('server-groups')
+        for server_group in server_groups:
+            if server_group['uuid'] in server_group_params:
+                server_group_ids.append(server_group['uuid'])
+                server_group_params.remove(server_group['uuid'])
+
+            elif server_group['name'] in server_group_params:
+                server_group_ids.append(server_group['uuid'])
+                server_group_params.remove(server_group['name'])
```

Comment:
This has the effect to only add the first server group with this name to the list of server group ids. Is this intended? I'd prefer either warning the user in this case or adding all matching server groups. What do you think?